cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,251 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import time
+import datetime
+try:
+    import boto3
+    import botocore
+except ImportError:
+    module.fail_json(msg='boto3 and botocore are required.')
+from distutils.version import LooseVersion
+if LooseVersion(botocore.__version__) < LooseVersion("1.3.14"):
+    module.fail_json(msg='botocore version 1.3.14 or above is required.')
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_nat_gateway
+short_description: Manage AWS VPC NAT Gateways
+description:
+  - Ensure the state of AWS VPC NAT Gateways based on their id, allocation and subnet ids.
+version_added: "2.1"
+options:
+  state:
+    description:
+      - Ensure NAT Gateway is present or absent
+    required: false
+    default: present
+    choices: ['present', 'absent']
+  nat_gateway_id:
+    description:
+      - The id AWS dynamically allocates to the NAT Gateway on creation
+    required: false
+  subnet_id:
+    description:
+      - The id of the subnet to create the NAT Gateway in
+    required: false
+  allocation_id:
+    description:
+      - The id of the elastic IP allocation
+    required: false
+  wait:
+    description:
+      - Wait for operation to complete before returning
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - How long to wait for an operation to complete before timing out
+    required: false
+    default: 300
+
+author: Jon Hadfield
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Ensure that a VPC NAT Gateway exists in a subnet by passing the subnet id and EIP allocation id
+ec2_vpc_nat_gateway:
+  state: present
+  subnet_id: subnet-a67643d1
+  allocation_id: eipalloc-a32baec6
+
+# Ensure that a VPC NAT Gateway exists in a subnet by passing the subnet id and
+# EIP allocation id but do not wait for operation to complete before continuing
+ec2_vpc_nat_gateway:
+  state: present
+  subnet_id: subnet-a67643d1
+  allocation_id: eipalloc-a32baec6
+  wait: false
+
+# Ensure that a VPC NAT Gateway identified by id does not exist
+ec2_vpc_nat_gateway:
+  state: absent
+  nat_gateway_id=nat-0d1e3a878585988f8
+'''
+RETURN = '''
+nat_gateway_id:
+    description: id of the VPC NAT Gateway
+    returned: when a VPC NAT Gateway is created
+    type: string
+    sample: "nat-0d1e3a878585988f8"
+'''
+
+
+def get_ec2_client(module=None):
+    """ Return an ec2 client """
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION "
```

Comment:
I don't like this particular message, e.g., you could also specify region in boto config or, per #12690 it could be the AWS_DEFAULT_REGION environment variable.